components/calculator.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Buttons from './buttons';
+import Display from './display';
+import calculate from '../logic/calculate';
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      previous,
+      operand,
+      current,
+    } = props;
+    this.state = {
+      total: previous,
+      operation: operand,
```

Comment:
- [ ] Please refactor `Calculator` component from a class-based component to a functional component 👍